tm2py/components/network/highway/highway_assign.py

```diff
@@ -190,16 +200,15 @@ def run(self):
 
                         # if need to update valuetoll, call set_tolls function
                         if update_toll_required:
-                            PrepareNetwork(controller=self._controller)._set_tolls(network=network, time_period=time)
+                            PrepareNetwork(controller=self._controller)._set_dynamic_tolls(network=network)
+                            PrepareNetwork(controller=self._controller)._calc_link_class_costs(network=network)
                             scenario.publish_network(network)
                         else:
-                            if dynamic_toll_iteration < 5:
+                            if dynamic_toll_iteration < 5 and valuetoll_links_exist:
```

Comment:
What is the purpose of the condition `dynamic_toll_iteration < 5`? It is skipping the assignment at iteration 5, which sounds incorrect to me.

Reply:
This handles the case when dynamic toll iteration < 5 (assignment only runs for 20 internal iterations), if no more links need to be updated, then run a regular assignment with 100 internal iterations.
For iteration 5 (last dynamic toll iteration), it will run 100 internal iterations at the first place.